core/iwasm/libraries/libc-wasi/libc_wasi_wrapper.c

```diff
@@ -969,9 +988,30 @@ wasi_poll_oneoff(wasm_exec_env_t exec_env, const wasi_subscription_t *in,
         || !validate_native_addr(nevents_app, sizeof(uint32)))
         return (wasi_errno_t)-1;
 
-    err = wasmtime_ssp_poll_oneoff(curfds, in, out, nsubscriptions, &nevents);
-    if (err)
-        return err;
+    int elapsed = 0;
+    /* in seconds */
+    const int timeout = get_timeout_for_poll_oneoff(in), time_quant = 1;
+    __wasi_subscription_t in_copy[nsubscriptions];
```

Comment:
This a [gcc extension](https://gcc.gnu.org/onlinedocs/gcc/Variable-Length.html); normally you can't use variable for the size as the size needs to be known at compile time. Better to allocate it explicitly on the heap.

Reply:
done